pkg/argparse/args.go

```diff
@@ -0,0 +1,288 @@
+
+	f.Var((*KVMap)(&a.Meta), "meta", "Specify the transient metainfo. Can be specified multiple times in key=value format.")
+	f.Var((*KVMap)(&a.MetaPersistent), "meta-persistent", "Specify the persistent metainfo. Can be specified multiple times in key=value format.")
+
+	f.BoolVar(&a.MetaBackward, "meta-backward", false, "Enable receiving backward metainfo from server.")
+
+	return f
+}
+
+func (a *Argument) PreJudge(f *flag.FlagSet) {
+	if len(os.Args) == 1 {
+		log.Println("Too few arguments. Try 'kitexcall -help' for more details.\n")
+		os.Exit(1)
+	}
+
+	if a.help {
+		Usage(f)
+		os.Exit(0)
+	}
+}
+
+func (a *Argument) ParseArgs() error {
+	f := a.buildFlags()
+	if err := f.Parse(os.Args[1:]); err != nil {
+		return errors.New(errors.ArgParseError, "Flag parse error: %v", err)
+	}
+
+	a.PreJudge(f)
+
+	log.Verbose = a.Verbose
+	log.Info("Parse args succeed")
+
+	if err := a.ValidateArgs(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *Argument) ValidateArgs() error {
+	var err error
+	if err = a.checkIDL(); err != nil {
+		return err
+	}
+
+	if err = a.checkData(); err != nil {
+		return err
+	}
+
+	if err = a.checkService(); err != nil {
+		return err
+	}
+
+	if err = a.checkTransport(); err != nil {
+		return err
+	}
+
+	log.Info("Args validate succeed")
+	return nil
+}
+
+func (a *Argument) checkTransport() error {
+	transport := strings.ToLower(a.Transport)
+	switch transport {
+	case strings.ToLower(config.TTHeader):
+		a.Transport = config.TTHeader
+	case strings.ToLower(config.Framed):
+		a.Transport = config.Framed
+	case strings.ToLower(config.TTHeaderFramed):
+		a.Transport = config.TTHeaderFramed
+	case "":
+		return errors.New(errors.ArgParseError, "Transport type is empty")
+	default:
+		return errors.New(errors.ArgParseError, "Transport type is invalid")
+	}
+	return nil
+}
+
+func (a *Argument) checkData() error {
+	// Data's priority is higher than File
+	if a.Data != "" {
+		return nil
+	}
+
+	if a.File == "" {
+		return errors.New(errors.ArgParseError, "At least one of Data or File must be specified")
+	}
+
+	if _, err := os.Stat(a.File); os.IsNotExist(err) {
+		return errors.New(errors.ArgParseError, "Input file does not exist")
+	}
+
+	// Check if the file is a JSON file.
+	if !strings.HasSuffix(a.File, ".json") {
+		return errors.New(errors.ArgParseError, "Input file must be in JSON format")
+	}
+	return nil
+}
+
+func (a *Argument) checkService() error {
+	// This can be modified when service discovery is supported.
+	if len(a.Endpoint) == 0 {
+		return errors.New(errors.ArgParseError, "At least one endpoint must be specified")
+	}
+
+	if a.Method == "" {
+		return errors.New(errors.ArgParseError, "Method name is required")
+	}
+
+	// Split the method string into service and method if '/'
+	parts := strings.Split(a.Method, "/")
+	switch len(parts) {
+	case 2: // If there is exactly one '/', it's treated as a separator
+		a.Service = parts[0]
```

Comment:
Also `IDLServiceName` here to avoid ambiguity

Reply:
done